# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/2d/DrawTargetSkia.cpp
# Commit: c3a89fd22e8c
# Full Hash: c3a89fd22e8c77473f56980900fe8638b4c2a993
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2025-07-22 09:32:06
# Description:
#   Bug 1966676 - Check for pattern surface before accessing offset. r=aosmond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D258034
# ==============================================================================

diff -r a4cb039768fc -r c3a89fd22e8c gfx/2d/DrawTargetSkia.cpp
--- a/gfx/2d/DrawTargetSkia.cpp	Mon Jul 21 17:26:55 2025 +0000
+++ b/gfx/2d/DrawTargetSkia.cpp	Mon Jul 21 17:30:59 2025 +0000
@@ -589,7 +589,9 @@
     case PatternType::SURFACE: {
       const SurfacePattern& pat = static_cast<const SurfacePattern&>(aPattern);
       Matrix offsetMatrix = pat.mMatrix;
-      offsetMatrix.PreTranslate(pat.mSurface->GetRect().TopLeft());
+      if (pat.mSurface) {
+        offsetMatrix.PreTranslate(pat.mSurface->GetRect().TopLeft());
+      }
       sk_sp<SkImage> image =
           GetSkImageForSurface(pat.mSurface, &aLock, aBounds, &offsetMatrix);
       if (!image) {
